<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="MatplotlibWidget" name="mplwidget"/>
    </item>
    <item>
     <widget class="QTabWidget" name="tabs">
      <property name="tabShape">
       <enum>QTabWidget::Triangular</enum>
      </property>
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Tab 1</string>
       </attribute>
       <widget class="QLabel" name="label">
        <property name="geometry">
         <rect>
          <x>140</x>
          <y>10</y>
          <width>81</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>RPM of output</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="setpoint_spinner">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>30</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>-100</number>
        </property>
        <property name="maximum">
         <number>2000</number>
        </property>
       </widget>
       <widget class="QPushButton" name="setpoint_button">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>60</y>
          <width>75</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Manual Speed</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="manual_spinner">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>91</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>-100</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
       <widget class="QCheckBox" name="auto_send_speed">
        <property name="geometry">
         <rect>
          <x>250</x>
          <y>30</y>
          <width>70</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Auto Send</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>PID Settings</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Kp</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Kd</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="ki_spinner_min">
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
          <property name="value">
           <double>10.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="kp_spinner_min">
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
          <property name="value">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="kd_spinner_min">
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QPushButton" name="pid_button">
          <property name="text">
           <string>Send Values</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Ki</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QSlider" name="ki_slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QSlider" name="kd_slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QSlider" name="kp_slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <widget class="QDoubleSpinBox" name="kp_spinner_max"/>
        </item>
        <item row="1" column="4">
         <widget class="QDoubleSpinBox" name="ki_spinner_max"/>
        </item>
        <item row="2" column="4">
         <widget class="QDoubleSpinBox" name="kd_spinner_max"/>
        </item>
        <item row="3" column="3">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Kd = 0.01 | Ki = 10.00 | Kd = 0.00</string>
          </property>
         </widget>
        </item>
        <item row="3" column="4">
         <widget class="QCheckBox" name="checkBox">
          <property name="text">
           <string>Auto Send</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>I/O</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Sending</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Recieving</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QTextBrowser" name="send_text_box">
          <property name="enabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QTextBrowser" name="recv_text_box">
          <property name="autoFormatting">
           <set>QTextEdit::AutoAll</set>
          </property>
          <property name="overwriteMode">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuConnections">
    <property name="title">
     <string>Connections</string>
    </property>
   </widget>
   <addaction name="menuConnections"/>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MatplotlibWidget</class>
   <extends>QWidget</extends>
   <header>matplotlibwidget</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
